core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/HasTrait.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.store.operation;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.commons.lang3.exception.CloneFailedException;
+
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.operation.serialisation.TypeReferenceImpl;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An Operation that will see if a Store has a given trait.
+ */
+@JsonPropertyOrder(alphabetic = true)
+@Since("2.0.0")
+@Summary("An Operation that will see if a Store has a given trait")
+public class HasTrait implements Operation, Output<Boolean> {
+    public static final boolean DEFAULT_CURRENT_TRAITS = true;
+
+    /**
+     * The currentTraits holds a boolean value, which if false
```

Comment:
Would it be useful to change this comment to reflect that it isn't used for returning a full list of supported traits for this operation but rather just comparing against either the store default or the schema current traits?